### PR TITLE
Add gardenlinux repo to base-test container

### PR DESCRIPTION
**Pull request by adaevitncar**
How to categorize this PR?
/kind enhancement
/area os
/os garden-linux

What this PR does / why we need it:
This PR adds the Garden Linux Repo to the base-test container in order to install some build tools that are built in the Gitlab Pipeline and published to the Garden Linux Repo. This makes them accessible in the Garden Linux base test.

Which issue(s) this PR fixes:
Fixes https://github.com/gardenlinux/gardenlinux/issues/1349

### CODE REVIEW

**Review by nanory — APPROVED** [review]

**Comment by Vincinator** [comment]
Tried to test this PR, but ran into an error

```
STEP 8/12: RUN echo "deb http://deb.debian.org/debian testing main contrib non-free" > /etc/apt/sources.list      && echo "deb [signed-by=$GARDENLINUX_MIRROR_KEY] https://repo.gardenlinux.io/gardenlinux $VERSION main" >> /etc/apt/sources.list      && apt-get update      && apt-get install -y --no-install-recommends           curl           unzip           ca-certificates           less           apt-transport-https           gnupg           python3-pip           python3-pytest           vim           procps           openssh-client           inetutils-ping           wget           iproute2           libguestfs-tools           qemu-system-x86           qemu-system-arm           qemu-efi-aarch64           shellcheck           build-essential           libpython3-dev
Ign:1 https://repo.gardenlinux.io/gardenlinux today InRelease
Get:2 http://deb.debian.org/debian testing InRelease [164 kB]
Get:3 http://deb.debian.org/debian testing/main amd64 Packages [8762 kB]
Get:4 http://deb.debian.org/debian testing/non-free amd64 Packages [117 kB]
Get:5 http://deb.debian.org/debian testing/contrib amd64 Packages [52.3 kB]
Ign:1 https://repo.gardenlinux.io/gardenlinux today InRelease
Ign:1 https://repo.gardenlinux.io/gardenlinux today InRelease
Err:1 https://repo.gardenlinux.io/gardenlinux today InRelease
  Certificate verification failed: The certificate is NOT trusted. The certificate issuer is unknown.  Could not handshake: Error in the certificate verification. [IP: 34.120.101.80 443]
Fetched 9094 kB in 7s (1294 kB/s)
Reading package lists...
W: https://repo.gardenlinux.io/gardenlinux/dists/today/InRelease: No system certificates available. Try installing ca-certificates.
W: http://deb.debian.org/debian/dists/testing/InRelease: The key(s) in the keyring /etc/apt/trusted.gpg.d/gardenlinux.asc are ignored as the file is not readable by user '_apt' executing apt-key.
W: https://repo.gardenlinux.io/gardenlinux/dists/today/InRelease: No system certificates available. Try installing ca-certificates.
W: https://repo.gardenlinux.io/gardenlinux/dists/today/InRelease: No system certificates available. Try installing ca-certificates.
W: https://repo.gardenlinux.io/gardenlinux/dists/today/InRelease: No system certificates available. Try installing ca-certificates.
W: Failed to fetch https://repo.gardenlinux.io/gardenlinux/dists/today/InRelease  Certificate verification failed: The certificate is NOT trusted. The certificate issuer is unknown.  Could not handshake: Error in the certificate verification. [IP: 34.120.101.80 443]
W: Some index files failed to download. They have been ignored, or old ones used instead.
```

**Comment by gyptazy** [comment]
Hey @Vincinator,

good catch. I got in touch w/ @adaevitncar and he's currently busy. Therefore, I provide a fix by commit dc3ed25 ad change from https to http for Garden Linux repository.

Reason:
We do not have any ca files (for x509 validation on https) but all gpg keys for the signed repositories. Therefore, we may switch to plain http instead. Further validation is done by GPG.

**Review by gyptazy — APPROVED** [review]

**Review by nanory — APPROVED** [review]

File: container/Makefile [hunk 1]
```diff
@@ -39,9 +39,11 @@ build-deb: build
 
 .PHONY: build-base-test
 build-base-test: needslim
+	cp ../gardenlinux.asc base-test/gardenlinux.asc
```
Comment (Vincinator) [inline container/Makefile:42]:
```suggestion
	cp -p ../gardenlinux.asc base-test/gardenlinux.asc
```

**Review by Vincinator — CHANGES_REQUESTED** [review]
Not sure why the permissions change when copying `gardenlinux.asc`  in the `container/Makefile`, but using the cp flag `-p` to preserve the attributes solved a follow up issue i had

File: container/Makefile (hunk 1 above)
Comment (gyptazy) [reply container/Makefile]: Hey @Vincinator,
thanks for your feedback. Can you provide more information about your issue? Maybe you already have an equal issue within the `build-image` target, where this file gets copied, too. However, preserving the file attributes should be fine here. Added within [51388fa](https://github.com/gardenlinux/gardenlinux/pull/1350/commits/51388fafeab1f64932277977ad6b256d096f64a3) (squashed).

Comment (Vincinator) [reply container/Makefile]:
```
W: http://deb.debian.org/debian/dists/testing/InRelease: The key(s) in the keyring /etc/apt/trusted.gpg.d/gardenlinux.asc are ignored as the file is not readable by user '_apt' executing apt-key.
W: http://deb.debian.org/debian-security/dists/testing-security/InRelease: The key(s) in the keyring /etc/apt/trusted.gpg.d/gardenlinux.asc are ignored as the file is not readable by user '_apt' executing apt-key.
W: http://deb.debian.org/debian/dists/testing-updates/InRelease: The key(s) in the keyring /etc/apt/trusted.gpg.d/gardenlinux.asc are ignored as the file is not readable by user '_apt' executing apt-key.
Reading package lists...
```

Added a line to `container/build-image/Dockerfile` to check file permissions of the gardenlinux.asc key
```
STEP 6/11: RUN ls -lah $GARDENLINUX_MIRROR_KEY
-rw-r----- 1 root root 1006 Nov 15 12:33 /etc/apt/trusted.gpg.d/gardenlinux.asc
```
But expected file permissions to be `644`

**Review by Vincinator — APPROVED** [review]

**Pull request merged**